src/particles/particle_creation.hpp

```diff
@@ -135,6 +139,250 @@ template <ParticleType particleType> struct ParticleCreationTraits {
+				amrex::Real vz = state_arr(i, j, k, HydroSystem<problem_t>::x3Momentum_index) / cell_density;
+				const amrex::Real t_ff = std::sqrt(3.0 * M_PI / (32.0 * C::Gconst * cell_density));
+				const amrex::Real particle_mass = cell_density * cell_volume * eps_ff * dt / t_ff;
+				const amrex::Real mass_low_mass_star = particle_mass * (1.0 - fstar_high);
+				double total_momx = 0.0;
+				double total_momy = 0.0;
+				double total_momz = 0.0;
+
+				for (int p_idx = 0; p_idx < num_particles; ++p_idx) {
+					auto &p = particles[p_idx]; // NOLINT
+
+					// Set particle ID and CPU
+					p.id() = pid_start + base_offset + p_idx;
+					p.cpu() = cpu_id;
+					p.rdata(birth_time_index) = current_time;
+
+					// Set particle evolution stage to 0 if it is a low mass star
+					// This gets changed if there is a high mass star in the cell
+					p.idata(evolution_stage_index) = p_idx;
+
+					// Low Mass particle position at cell center
+					p.pos(0) = plo[0] + (i + 0.5) * dx[0];
+					p.pos(1) = plo[1] + (j + 0.5) * dx[1];
+					p.pos(2) = plo[2] + (k + 0.5) * dx[2];
+
+					// Low Mass particle mass and velocity
+					p.rdata(mass_idx) = mass_low_mass_star;
+					p.rdata(mass_idx + 1) = vx;
+					p.rdata(mass_idx + 2) = vy;
+					p.rdata(mass_idx + 3) = vz;
+
+					p.rdata(birth_time_index + 1) = LONG_MAX;
+					if (p_idx > 0) {
+						double sigma_sq_x = NAN;
+						double sigma_sq_y = NAN;
+						double sigma_sq_z = NAN;
+						double numx = 0.0;
+						double numy = 0.0;
+						double numz = 0.0;
+						double denominator = 0.0;
+						double vx_adj = NAN;
+						double vy_adj = NAN;
+						double vz_adj = NAN;
+						double rho_adj = NAN;
+						// Get the average velocity from the velocity dispersion of the surrounding cells
+						// We use the velocity dispersion of the surrounding cells to get the velocity of the high mass star...
+						//... from a log normal distribution
+						for (int ii = i - 1; ii <= i + 1; ++ii) {
+							for (int jj = j - 1; jj <= j + 1; ++jj) {
+								for (int kk = k - 1; kk <= k + 1; ++kk) {
+
+									vx_adj = (state_arr(ii, jj, kk, HydroSystem<problem_t>::x1Momentum_index)) /
+										 state_arr(ii, jj, kk, HydroSystem<problem_t>::density_index);
+									vy_adj = (state_arr(ii, jj, kk, HydroSystem<problem_t>::x2Momentum_index)) /
+										 state_arr(ii, jj, kk, HydroSystem<problem_t>::density_index);
+									vz_adj = (state_arr(ii, jj, kk, HydroSystem<problem_t>::x3Momentum_index)) /
+										 state_arr(ii, jj, kk, HydroSystem<problem_t>::density_index);
+									rho_adj = state_arr(ii, jj, kk, HydroSystem<problem_t>::density_index);
+									numx += rho_adj * (vx_adj - (vx)) * (vx_adj - (vx));
+									numy += rho_adj * (vy_adj - (vy)) * (vy_adj - (vy));
+									numz += rho_adj * (vz_adj - (vz)) * (vz_adj - (vz));
+
+									denominator += rho_adj;
+								}
+							}
+						}
+						sigma_sq_x = numx / denominator;
```

Comment:
For better readability, you can do the following since `sigma_sq_x` is only assigned once. Similarly for other variables.
```suggestion
						const double sigma_sq_x = numx / denominator;
```